src/security/hapi-auth-validation.js

```diff
@@ -36,32 +49,44 @@ module.exports = (options, server, request, userId, callback) => {
       }
 
       const requestParams = authPlugin.getParams ? authPlugin.getParams(request) : {}
-      const buildParams = Object.assign({}, user, requestParams)
+      const buildParams = Object.assign({}, { organizationId: currentUser.organizationId }, requestParams)
 
       let resource = authPlugin.resource
       if (!resource) {
         const resourceType = route.path.split('/')[2]
         if (!authConfig.resources[resourceType]) {
-          return callback(null, false, { error: 'Resource builder not found' })
+          return next(null, false, { error: 'Resource builder not found' })
         }
 
         const resourceBuilder = authConfig.resources[resourceType]
-        resource = resourceBuilder && resourceBuilder(buildParams) || null
-      }
 
-      const params = {
-        userId,
-        action: authPlugin.action,
-        resource
-      }
+        if (resourceType === 'users' && buildParams.userId) {
```

Comment:
Could we extract the resource building process into its own function?

My idea is to have something like

```
if (!resource) {
  ...
  return generateResources({...}, {err, resources} => {
    proceed(resources)
  })
}

proceed([resources])
```